README.md

```diff
@@ -0,0 +1,99 @@
+    - **appId**: Service principal application id.Configured using environment variable **LOG4J2_ADX_APP_ID**
+    - **appKey**: Service principal application secret.Configured using environment variable **LOG4J2_ADX_APP_KEY**
+    - **appTenant**: Tenant for the Service principal.Configured using environment variable **LOG4J2_ADX_TENANT_ID**
+    - **dbName**: Database name.Configured using environment variable **LOG4J2_ADX_DB_NAME**
+    - **tableName**: Table name for ingesting the logs
+    - **logTableMapping**: Mapping defined in the database to map the log data
+    - **mappingType**: json (or) csv is currently supported. Defaults to **_csv_**
+    - **flushImmediately**: Boolean indicator to flush the logs immediately. Defaults to **_false_**. Note that making
+      this true may cause additional load on the cluster
+    - **proxyUrl**: Proxy url in case application is hosted behind a proxy
+
+- To attempt retries in case of ingestion failures, retransmission is attempted with the following configuration. 3
+  retries are attempted to ingest the logs. In the event that the file cannot be ingested it gets moved to the backout
+  directory in the same path defined in fileName
+
+    - **backOffMinMinutes**: Min minutes to back off in the event that ingestion fails
+    - **backOffMaxMinutes**: Max minutes to back off in the event that ingestion fails
+
+```xml
+<Configuration status="WARN">
+    <Appenders>
+        <RollingFile name="ADXRollingFile" fileName="<fileName>"
+                     filePattern="<filePattern>">
+            <KustoStrategy
+                    clusterPath="https://ingest-<cluster>.kusto.windows.net"
+                    appId=""
+                    appKey=""
+                    appTenant=""
+                    dbName=""
+                    tableName=""
+                    logTableMapping=""
+                    mappingType=""
+                    flushImmediately=""
+                    proxyUrl=""
+                    backOffMinMinutes=""
+                    backOffMaxMinutes=""
+            />
+            <CsvLogEventLayout delimiter="," quoteMode="ALL"/>
+            <Policies>
+              <!-- Recommended size is 4 MB -->
+                <SizeBasedTriggeringPolicy size="4 MB"/>
+                <TimeBasedTriggeringPolicy interval="5" modulate="true"/>
+            </Policies>
+        </RollingFile>
+    </Appenders>
+    <Loggers>
+        <Root level="debug" additivity="false">
+            <AppenderRef ref="ADXRollingFile"/>
+        </Root>
+    </Loggers>
+</Configuration>
+```
+
+How to build:
+----------------------
+In power-shell the following can be set
+
+```sh
+$env:LOG4J2_ADX_DB_NAME="<db-name>"
+$env:LOG4J2_ADX_ENGINE_URL="https://<cluster>.kusto.windows.net"
+$env:LOG4J2_ADX_TENANT_ID="<tenant-id>"                   
+$env:LOG4J2_ADX_INGEST_CLUSTER_URL="https://ingest-<cluster>.kusto.windows.net"

```

Comment:
I believe there was a recent change that allowed us to deterministically move between the ingestion and query endpoints.
Accordingly, we made this change in Java: https://github.com/Azure/azure-kusto-java/pull/268